File: tests/TestSample.java
// Test sample for CodeSentinel — intentional bugs, style issues, and security vulnerabilities

import java.sql.*;
import java.io.*;

public class TestSample {

    // Security: hardcoded password (HIGH)
    private static String DB_PASSWORD = "hardcoded_password_123";

    // Security: SQL injection via string concatenation (HIGH)
    public static String getUserById(Connection conn, String userId) throws SQLException {
        String query = "SELECT * FROM users WHERE id = " + userId;
        Statement stmt = conn.createStatement();
        ResultSet rs = stmt.executeQuery(query);
        if (rs.next()) {
            return rs.getString("username");
        }
        return null;
    }

    // Bug: resource leak — FileWriter never closed
    // Bug: silent catch — exception swallowed
    public static void writeToFile(String filename, String content) {
        try {
            FileWriter fw = new FileWriter(filename);
            fw.write(content);
        } catch (IOException e) {
            // exception silently ignored
        }
    }

    // Bug: no guard against division by zero
    public static int divide(int a, int b) {
        return a / b;
    }

    // Style: missing Javadoc; method name should be camelCase (processData)
    public static void ProcessData(String[] items) {
        for (int i=0; i<items.length; i++) {  // Style: missing spaces around =
            System.out.println(items[i]);
        }
    }

    public static void main(String[] args) {
        int result = divide(10, 0);  // Bug: throws ArithmeticException at runtime
        System.out.println(result);
    }
}

File: tests/TestSample.cs
// Test sample for CodeSentinel — intentional bugs, style issues, and security vulnerabilities

using System;
using System.Data.SqlClient;

public class TestSample
{
    // Security: hardcoded credentials in connection string (HIGH)
    private string connectionString = "Server=localhost;Password=admin123";

    // Security: SQL injection via string concatenation (HIGH)
    public string GetUser(string userId)
    {
        string query = "SELECT * FROM Users WHERE Id = " + userId;
        using (SqlConnection conn = new SqlConnection(connectionString))
        {
            SqlCommand cmd = new SqlCommand(query, conn);
            conn.Open();
            return cmd.ExecuteScalar()?.ToString();
        }
    }

    // Bug: no guard against division by zero — throws DivideByZeroException
    public int Divide(int a, int b)
    {
        return a / b;
    }

    // Bug: off-by-one — i <= items.Length causes IndexOutOfRangeException
    public void ProcessItems(string[] items)
    {
        for (int i = 0; i <= items.Length; i++)
        {
            Console.WriteLine(items[i]);
        }
    }

    // Style: method name should be PascalCase (BadMethod)
    // Style: missing spaces around operators and after 'if'
    public void badMethod()
    {
        var x=1;
        var unused="never used";  // Style: unused variable
        if(x==1)
            Console.WriteLine("bad style");
    }
}
